extensions/wikia/Captcha/Module/BaseCaptcha.class.php

```diff
@@ -0,0 +1,235 @@
+<?php
+
+namespace Captcha\Module;
+
+use Captcha\Factory;
+use Wikia\Logger\WikiaLogger;
+
+/**
+ * Class BaseCaptcha
+ *
+ * @package Captcha\Module
+ */
+abstract class BaseCaptcha {
+
+	public $action;
+
+	public $actionDetail;
+
+	/**
+	 * Called to determine what field to look for when validating if the captcha has loaded
+	 *
+	 * @return string
+	 */
+	abstract function checkCaptchaField();
+
+	function isValid() {
+		return true;
+	}
+
+	public function getCaptcha() {
+		$a = mt_rand( 0, 100 );
+		$b = mt_rand( 0, 10 );
+
+		/* Minus sign is used in the question. UTF-8,
+		   since the api uses text/plain, not text/html */
+		$op = mt_rand( 0, 1 ) ? '+' : '−';
+
+		// No space before and after $op, to ensure correct
+		// directionality.
+		$test = "$a$op$b";
+		$answer = ( $op == '+' ) ? ( $a + $b ) : ( $a - $b );
+		return [
+			'question' => $test,
+			'answer' => $answer
+		];
+	}
+
+	public function addCaptchaAPI( &$resultArr ) {
+		$captcha = $this->getCaptcha();
+		$index = $this->storeCaptcha( $captcha );
+		$resultArr['captcha']['type'] = 'simple';
+		$resultArr['captcha']['mime'] = 'text/plain';
+		$resultArr['captcha']['id'] = $index;
+		$resultArr['captcha']['question'] = $captcha['question'];
+	}
+
+	/**
+	 * Insert a captcha prompt into the edit form.
+	 * This sample implementation generates a simple arithmetic operation;
+	 * it would be easy to defeat by machine.
+	 *
+	 * Override this!
+	 *
+	 * @return string HTML
+	 */
+	public function getForm() {
+		$captcha = $this->getCaptcha();
+		$index = $this->storeCaptcha( $captcha );
+
+		return "<p><label for=\"wpCaptchaWord\">{$captcha['question']}</label> = " .
+		\Xml::element( 'input', [
+			'name' => 'wpCaptchaWord',
+			'id'   => 'wpCaptchaWord',
+			'tabindex' => 1,
+		] ) . // tab in before the edit textarea
+		"</p>\n" .
+		\Xml::element( 'input', [
+			'type'  => 'hidden',
+			'name'  => 'wpCaptchaId',
+			'id'    => 'wpCaptchaId',
+			'value' => $index,
+		] );
+	}
+
+	/**
+	 * Show a message asking the user to enter a captcha on edit
+	 * The result will be treated as wiki text
+	 *
+	 * @param string $action Action being performed
+	 *
+	 * @return string
+	 */
+	public function getMessage( $action ) {
+		$name = 'captcha-' . $action;
```

Comment:
Yah, this same method in one of the child classes has those enumerated, I can add it here as well (though technically we should never get here)